DOC Improve contributing instructions (#2447)
pyodide/pyodide: docs/development/contributing.md

## Development Workflow

To contribute code, see the following steps,

1. Fork the Pyodide repository [https://github.com/pyodide/pyodide](https://github.com/pyodide/pyodide) on Github.

> 2. If you are on Linux, you can skip this step. On Windows and MacOS you have a

Reviewer: It's still recommended to use docker in linux though?

Author: Yeah, but we already say that in the section about building from sources. So I think it's fine.

>    choice between, manually installing Docker,

Reviewer: ```suggestion
   choice. The first option is to manually install Docker:
```

   - on MacOS follow [these instructions](https://docs.docker.com/desktop/mac/install/)
   - on Windows, [install WSL
     2](https://docs.microsoft.com/en-us/windows/wsl/install), then Docker.
     Note that Windows filesystem access from WSL2 is very slow and should
     be avoided when building Pyodide.

>    The other alternative is to use a service that provides a Linux

Reviewer: ```suggestion
   The second option is to use a service that provides a Linux
```

>    development environment, such as,

Reviewer: There should either be a colon or no punctuation here and in other places before display code.
```suggestion
   development environment, such as
```

   - [Github Codespaces](https://github.com/features/codespaces)
   - [gitpod.io](https://gitpod.io)
   - or a remote Linux VM with SSH connection.

> 3. Clone your fork of Pyodide,

Reviewer: ```suggestion
3. Clone your fork of Pyodide
```
